File: JULY-2025/28-07-2025/p2044.c
int backtrack(int* nums, int n, int i, int curOr, int tarOr) {
    if (curOr == tarOr) return 1 << (n - i);
    int count = 0;
    for (; i < n; ++i) count += backtrack(nums, n, i+1, curOr | nums[i], tarOr);
    return count;
}

int countMaxOrSubsets(int* nums, int n) {
    int maxOr = 0;
    for (int i = 0; i < n; ++i) maxOr |= nums[i];
    return backtrack(nums, n, 0, 0, maxOr);
}

File: JULY-2025/28-07-2025/p2044.php
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function countMaxOrSubsets($nums) {
         $maxiOr = PHP_INT_MIN;
    $c = 0;
    $n = count($nums);

    for ($i = 1; $i < (1 << $n); $i++) {
        $or = 0;
        for ($j = 0; $j < $n; $j++) {
            if (($i & (1 << $j)) != 0) {
                $or |= $nums[$j];
            }
        }
        if ($or > $maxiOr) {
            $maxiOr = $or;
            $c = 0;
        }
        if ($or == $maxiOr) {
            $c += 1;
        }
    }
    return $c;
    }
}